### PR TITLE
Inbox notifications on overview screen

### LINKED ISSUE #1662: Inbox notifications in overview screen (as part of WCPay overview v2)
This covers the work needed to display wc pay specific notifications in the overview screen. (For detailed designs and discussion see paJDYF-18j-p2). **Note that the designs are out of date now because the intent is to display this inline on the page. If you want a design to work from, assume the look and feel should match the inline notifications in wc-admin.**

I think we need to do some exploration work or a spike to determine how this will be integrated with or share code from the woocommerce-admin notifications.

cc @pmcpinto 

Requirements

- [ ] Display wc-pay specific notifications inline in the new wc pay overview screen
- [ ] TBD analytics required?

### PR DESCRIPTION
**Pull request by louwie17**
Fixes #1662 

#### Changes proposed in this Pull Request

Adds a InboxPanel component to the overview screen that display's woocommerce payment notes.

This adds a InboxPanel component within the overview directory to be displayed on the overview screen.
In order for us to only query `woocommerce-payments` notes, I also had to add a few filters to the `class-wc-payments.php`, which allows us to query by `source`.
I also added a helper function for updating the woocommerce_meta field in the current user object, this is used for showing if a note is new or not - storing the timestamp in the meta field. We do the same in **WooCommerce > Home**.

wcpay-notes

#### Testing instructions
- Load this branch
- Go to **Payments > Overview**
- It should briefly show a loading inbox note placeholder
- Two notes should be present - these are the only woocommerce-payments notes (at-least for me) - `Set up refund policy` & `Force secure checkout` (as shown in the gif above) 
- All note interactions should work (dismissing one/all, and clicking on the action).
- With the `debug` field set to `wc-admin:*` in localstorage you should see the `wcpay_inbox_note_view` when scrolling down for each note.

*

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/overview/inbox-notifications/index.js [hunk 1]
```diff
@@ -0,0 +1,315 @@
+/* eslint-disable camelcase */
```
Comment (louwie17) [inline client/overview/inbox-notifications/index.js:1]: It seems that a large percentage of js files in this repo make use of the above `eslint-disable camelcase`, maybe we should just disable it? https://github.com/Automattic/woocommerce-payments/blob/develop/.eslintrc#L32

**Comment by octaedro** [comment]
Hi @louwie17,
I'm testing this PR and I find a couple of things that aren't working as expected.
- The `Dismiss all messages` button is not dismissing the notes with source `woocommerce-payments` but is dismissing notes that are not visible under: `Payments` > `Overview`. I'm not sure whether that is the expected behavior but it feels odd for me.
- After actioned a note with source `woocommerce-payments`, the note never changes its status to `actioned`. So it's still visible after refreshing the page.

Please let me know if I missed a step or did something wrong while testing.

File: client/overview/inbox-notifications/index.js [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+			/>
+		);
+	}
+
+	const onDismiss = ( note, type ) => {
+		setDismiss( { note, type } );
+	};
+
+	const closeDismissModal = async ( confirmed = false ) => {
+		const noteNameDismissAll = 'all' === dismiss.type ? true : false;
```
Comment (octaedro) [inline client/overview/inbox-notifications/index.js:210]: Nitpick: I think that this is the same as:
`const noteNameDismissAll = 'all' === dismiss.type;`

**Review by octaedro — COMMENTED** [review]
Good job @louwie17! Code looks good, just left a small comment and I had the mentioned issues when I was testing the PR.

File: client/overview/inbox-notifications/index.js (hunk 2 above)
Comment (louwie17) [reply client/overview/inbox-notifications/index.js]: Thanks, yeah that is nicer, updated in: https://github.com/Automattic/woocommerce-payments/pull/2053/commits/c9bc8d4413d9618c80a8439d086ca3de6ece02ea

**Comment by louwie17** [comment]
I appreciate the review @octaedro 😄 

> The Dismiss all messages button is not dismissing the notes with source woocommerce-payments but is dismissing notes that are not visible under: Payments > Overview. I'm not sure whether that is the expected behavior but it feels odd for me.

This was indeed wrong behaviour, nice catch!! I fixed this in https://github.com/Automattic/woocommerce-payments/commit/c9bc8d4413d9618c80a8439d086ca3de6ece02ea by only updating the displayed notes. I used the batch update function as the delete all didn't support any query params.

> After actioned a note with source woocommerce-payments, the note never changes its status to actioned. So it's still visible after refreshing the page.

This is technically expected, as the note actions [are set to `unactioned` upon click](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/notes/class-wc-payments-notes-set-https-for-checkout.php#L61), I asked in Slack ( p1623775742051000-slack-CGGCLBN58 ) if this is expected behaviour or a mis-understanding of the notes API.
I don't think this should be a blocker, and can update this in a separate PR if I don't hear back quickly.

@octaedro another look would be appreciated 😃

**Review by octaedro — APPROVED** [review]
Now everything is working as expected. Nice job @louwie17 :shipit:

File: client/overview/inbox-notifications/index.js (hunk 1 above)
Comment (mattsherman) [reply client/overview/inbox-notifications/index.js:1]: Did you ever follow up on this? I thought I remembered seeing something about this elsewhere?

Comment (louwie17) [reply client/overview/inbox-notifications/index.js:1]: I had left a comment in Slack, but didn't get a huge response there, aside from one person.
So I didn't change anything "yet"

**Pull request merged**